test/files/jvm/future-spec/FutureTests.scala

```diff
@@ -39,7 +41,8 @@ class FutureTests extends MinimalScalaTest {
       val ms = new concurrent.TrieMap[Throwable, Unit]
       implicit val ec = scala.concurrent.ExecutionContext.fromExecutor(new java.util.concurrent.ForkJoinPool(), {
         t =>
-        ms += (t -> ())
+        val u = ()
+        ms += (t -> u)
```

Comment:
This is clearly a parser bug. `42 -> ()` is `42.$minus$greater()`.

Note that `t -> {}` works. `42 -> {}` is parsed `42.$minus$greater(())`. What a hard-working parser.

Reply:
I forgot to say that we're so concerned with `()` insertion that we've become inured to `()` deletion.

Reply:
In terms of the adoption, or lack thereof, parsing `42 -> ()` as `42.$minus$greater()` sounds like exactly what we want.
The ability to parse this as `42.$minus$greater(())` I think is more related to auto-tupling. It might be good to find out what Dotty does here.